### DISCUSSION TITLE
Set "Primary Site Collection Administrator"/Owner

### DISCUSSION THREAD

**Discussion by mark3grahams** [pnp/powershell#2284] [Q&A]
Set-PnPTenantSite -Identity "https://contoso.sharepoint.com/sites/sales" -Owners "user@contoso.onmicrosoft.com"
This only adds additional "Site Collection Administrators" but leaves the owner untouched.

Set-SPOSite -Identity "https://contoso.sharepoint.com/sites/sales" -Owner "user@contoso.onmicrosoft.com"
This does what I want but I use a certificate for authentication and Connect-SPOService does not appear to support this yet.

Is there any other way to set the owner?

**Comment by Nick-MAC** [comment]
I would also  like an answer to this... I've tried the following but apparently it is a read only property. Guys, surely there is an answer here.

$site = Get-pnptenantsite -url "siteurlhere"
$Site.Owner = "NewOwner"
$Site.Update()

**Comment by BaronSparky** [comment]
Have you tried, after connecting to the site:
`Get-PnPGroup -AssociatedOwnerGroup |  Add-PnPGroupMember -LoginName "<user>@<tenant>.onmicrosoft.com"`

**Reply by Nick-MAC** [reply]
I have not. I saw this but assumed this was specifically for managing the associated owners group on a Modern/Team site.

**Reply by BaronSparky** [reply]
> I have not. I saw this but assumed this was specifically for managing the associated owners group on a Modern/Team site.

Still works for Classic site collections. This is a "classic" team site:
![image](https://user-images.githubusercontent.com/11838512/199437582-aa83b773-7085-4c41-8dd9-e92ae85d8dbf.png)

**Reply by Nick-MAC** [reply]
Nice, I'll give this a try. Thanks!

**Reply by shcrist** [reply]
This would only get the site owners group to add members to that group. That would not set the owner of the site.

**Comment by AndersRask** [comment]
For Site collection admin use the appropriately named Add-PnPSiteCollectionAdmin while connected to the site

`Add-PnPSiteCollectionAdmin -Owners @("user1@contoso.onmicrosoft.com", "user2@contoso.onmicrosoft.com")`

**Reply by Nick-MAC** [reply]
I am aware of this command already. This only adds site collection administrators. I am trying to update the primary owner.

In short, I am looking for the pnp equivalent of Set-SPOsite -url "siteurl" -owner "newsiteowner"

Thanks for the suggestion but this does not get me what I'm looking for.

**Reply by AndersRask** [reply]
Ok, buit your title for the question was Set "Primary Site Collection Administrator"/Owner 

Those are two different things. Use Add-PnPGroupMember to add owners to owners group and use Add-PnPSiteCollectionAdmin to add SC admins.

**Reply by shcrist** [reply]
Actually, they're not two different things. You're confusing the site owner with being a member of the site owners group. The site owner is the primary administrator for a site and is part of the site collection administrators group. If you get the site object and look at the Owner property, you'll see the account of the owner. In SPO Admin, you click on a site, select permissions, and then select the manage link. The SCAs are listed and one of them will be the Primary Admin or Owner. That's the account that needs to be set here.

I have the same problem and hope you find a solution. I've tried the same methods as you and have not found a way that works.

**Comment by mark3grahams** [comment]
So, there is still no way to do this?

**Reply by shcrist** [reply]
It was added in the latest release. Add-PnPSiteCollectionAdmin -PrimarySiteCollectionAdmin "<user>"

**Reply by Nick-MAC** [reply]
Nice! About time

**Reply by mark3grahams** [reply]
And that is Posh 7+ only for the latest update.

**Reply by ryszardcopa** [reply]
PnP.Powershell 2.2.0 seems does not have it. Which version you mean?

**Reply by shcrist** [reply]
It's in version 2.2.0. I just used it on Monday. See https://pnp.github.io/powershell/cmdlets/Add-PnPSiteCollectionAdmin.html example 4.

**Reply by ryszardcopa** [reply]
My bad. I switched to PS5. I am sorry. Thanks for quick response.
